### PR TITLE
 Replace GenMod with own implementation for inheritance calculation.

### PR DESCRIPTION
**Pull request by bartcharbon**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,82 @@
...
-usage: java -jar vcf-inheritance-matcher.jar -v
- -v,--version   Print version.
+       <arg>] [-pb <arg>] [-np <arg>] [-c] [-f] [-d]
+ -i,--input <arg>            Input VCF file (.vcf or .vcf.gz).
+ -o,--output <arg>           Output VCF file (.vcf or .vcf.gz).
+ -pd,--pedigree <arg>        Comma-separated list of pedigree files
+                             (.ped).
+ -pb,--probands <arg>        Comma-separated list of proband sample
+                             identifiers.
+ -np,--nonpenetrance <arg>   File containing a list of non penetrance
```
Comment (dennishendriksen) [inline README.md:29]: terminology: shouldn't this be reduced penetrance instead of non penetrance?

File: README.md [hunk 2]
```diff
@@ -1,35 +1,82 @@
...
+                             genes (.tsv), first column is assumed to
+                             contain the genes.
+ -c,--annotateCompounds      Annotate variant/sample combinations with a
```
Comment (dennishendriksen) [inline README.md:32]: suggestion: consistency with our other CLIs: annotate_compounds instead of annotateCompounds

Comment (dennishendriksen) [reply README.md:32]: is there a reason why the user would not want to have these annotations?

File: src/test/resources/nonPen.tsv [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Gene
```
Comment (dennishendriksen) [inline src/test/resources/nonPen.tsv:1]: not all columns have a column header. can we add column headers for all columns to make it a 'more standard' .tsv?

File: src/test/resources/nonPen.tsv [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+ID1
```
Comment (dennishendriksen) [inline src/test/resources/nonPen.tsv:2]: question: is it allowed to add an additional column with the gene symbol (that is not used by the tool, but can be informative for the user?)

File: src/test/resources/expected_noVEPinheritance.vcf [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+##FORMAT=<ID=VIG,Number=.,Type=String,Description="Genes with an inheritance match.">
+##FORMAT=<ID=VIM,Number=1,Type=Integer,Description="Inheritance Match status.">
+##FORMAT=<ID=VIS,Number=.,Type=String,Description="An enumeration of possible sub inheritance modes like e.g. compound, non penetrance.">
+##INFO=<ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|SYMBOL|Gene">
+##INFO=<ID=Compounds,Number=.,Type=String,Description="List of compound pairs for this variant.The list is splitted on ',' family id is separated with compoundswith ':'. Compounds are separated with '|'.">
+##INFO=<ID=GeneticModels,Number=.,Type=String,Description="':'-separated list of genetic models for this variant.">
+##INFO=<ID=ModelScore,Number=.,Type=String,Description="PHRED score for genotype models.">
+##PREFERRED=Flags a transcript as preferred
+##Software=<ID=genmod,Version=3.7.3,Date="2020-11-19 08:13",CommandLineOptions="processes=1 keyword=SYMBOL family_type=ped family_file=<_io.TextIOWrapper name='./pedigree_complex.ped' mode='r' encoding='UTF-8'> variant_file=./workdir/splitted.vcf context=<click.core.Context object at 0x7fcfa0a0cf60>">
```
Comment (dennishendriksen) [inline src/test/resources/expected_noVEPinheritance.vcf:16]: Suggestion: remove all references to genmod

File: src/test/resources/expected_noVEPinheritance.vcf [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+##FORMAT=<ID=VIC,Number=1,Type=String,Description="Inheritance Compound status.">
+##FORMAT=<ID=VID,Number=1,Type=Integer,Description="Inheritance Denovo status.">
+##FORMAT=<ID=VIG,Number=.,Type=String,Description="Genes with an inheritance match.">
+##FORMAT=<ID=VIM,Number=1,Type=Integer,Description="Inheritance Match status.">
+##FORMAT=<ID=VIS,Number=.,Type=String,Description="An enumeration of possible sub inheritance modes like e.g. compound, non penetrance.">
+##INFO=<ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|SYMBOL|Gene">
+##INFO=<ID=Compounds,Number=.,Type=String,Description="List of compound pairs for this variant.The list is splitted on ',' family id is separated with compoundswith ':'. Compounds are separated with '|'.">
+##INFO=<ID=GeneticModels,Number=.,Type=String,Description="':'-separated list of genetic models for this variant.">
+##INFO=<ID=ModelScore,Number=.,Type=String,Description="PHRED score for genotype models.">
+##PREFERRED=Flags a transcript as preferred
```
Comment (dennishendriksen) [inline src/test/resources/expected_noVEPinheritance.vcf:15]: Suggestion: remove all references to PREFERRED transcripts (plugin will be removed in VIP v4.0.0)

File: README.md (hunk 2 above)
Comment (bartcharbon) [reply README.md:32]: is removed already, only forgot to remove it from the docs as well

File: README.md (hunk 1 above)
Comment (bartcharbon) [reply README.md:29]: depends on who you ask I guess, this is the term our department uses I think

File: src/test/resources/nonPen.tsv (hunk 4 above)
Comment (bartcharbon) [reply src/test/resources/nonPen.tsv:2]: any additional columns are ignored, so you can add everything you like

**Comment by bartcharbon** [comment]
I seem to be overwriting my results without using "--force"

**Comment by bartcharbon** [comment]
max_records seems to be broken

**Pull request merged**